pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -255,6 +256,9 @@ func (m *Multicluster) initializeCluster(cluster *multicluster.Cluster, kubeCont
 					configStore, options.XDSUpdater,
 					serviceentry.WithClusterID(cluster.ID),
 					serviceentry.WithNetworkIDCb(kubeRegistry.Network))
+				if m.serviceEntryController != nil {
+					m.serviceEntryController.AppendServiceHandler(kubeRegistry.ServiceEntryHandler)
+				}
```

Comment:
i think we need to append this service entry handler regardless of `features.EnableK8SServiceSelectWorkloadEntries` (see line 247)

Reply:
addressed in https://github.com/istio/istio/pull/45621/commits/3d10afbdc7a5c1be50ac7894899404c78bbb355f

Reply:
addressed in https://github.com/istio/istio/pull/45621/commits/7055f9065d95eb1db7e5b07450f6d5288bd7fb6f